src/render/mesh.rs

```diff
@@ -226,15 +230,33 @@ pub struct Link {
 }
 
 impl Link {
+    /// Match mesh attributes against shader inputs, produce a mesh link.
+    /// Exposed to public to allow external `Batch` implementations to use it.
+    pub fn new<R: Resources>(mesh: &Mesh<R>, pinfo: &device::shade::ProgramInfo)
+                             -> Result<Link, Error> {
+        let mut indices = Vec::new();
+        for sat in pinfo.attributes.iter() {
+            match mesh.attributes.iter().enumerate()
```

Comment:
It is `N^2`, yes, and we are doing it when constructing a batch, so it doesn't happen during rendering.